src/model/resolvable.ts

```diff
@@ -18,9 +18,13 @@ import {Feature, ScannedFeature} from './feature';
 /**
  * A ScannedFeature that needs to resolve other Features to build its final
  * representation.
+ *
+ * TODO(aomarks) Is it right that I made this an abstract class? ScannedFeature
```

Comment:
If Resolvable remains an interface, then the compiler requires me to re-declare all the properties from the ScannedFeature definition (description, sourceRange, etc.), and also I can't call the ScannedFeature constructor -- because I'm no longer extending an abstract class, I'm implementing an interface derived from it, so all the abstract provisions are lost. Which seems to defeat the purpose of having an abstract class in the hierarchy?

@rictic Sorry, I'm not sure if you're advocating for the interface or the abstract class?